lib/raxx/middleware.ex

```diff
@@ -147,6 +147,36 @@ defmodule Raxx.Middleware do
   """
   @callback process_info(any(), state(), inner_server :: Server.t()) :: next()
 
+  defmacro __using__(_options) do
+    quote do
```

Comment:
I'm pretty sure this is missing some `defoverridable`s. If I update the test below like this:

```elixir
  defmodule DefaultMiddleware do
    use Middleware

    def process_head(message, state, inner_server) do
      IO.inspect {message, state, inner_server}
      {[], state, inner_server}
    end
  end
```

Instead of an extra inspect output and a failed test, I get a compilation warning:

```
warning: this clause cannot match because a previous clause at line 47 always matches
  test/raxx/stack_test.exs:49
```

Reply:
Also, both `GenServer` and `Raxx.Server` do `@before_compile unquote(__MODULE__)`. I'd be curious why that is and see if we need to add it here too.

Reply:
Definitely missing a defoverridable. Will re-vist and get this PR actually up to scratch.

Reply:
Checked, we don't need a `@before_compile` here. In both cases you mentioned that is just to handle some warnings